lib/evm_database_ops.rb

```diff
@@ -28,47 +29,61 @@ def self.database_size(opts)
   end
 
   def self.backup(db_opts, connect_opts = {})
-    # db_opts:
-    #   :dbname => 'vmdb_production',
-    #   :username => 'root',
-    #   :local_file => "/tmp/backup_1",      - Backup locally to the file specified
+    uri = run_db_opt_for_new_uri(:backup, db_opts, connect_opts) do |database_opts|
+      PostgresAdmin.backup(database_opts)
+    end
+    _log.info("[#{db_opts[:dbname]}] database has been backed up to file: [#{uri}]")
+    uri
+  end
 
-    # connect_opts:
-    #   :uri => "smb://dev005.manageiq.com/share1",
-    #   :username => 'samba_one',
-    #   :password => 'Zug-drep5s',
-    #   :remote_file_name => "backup_1",     - Provide a base file name for the uploaded file
+  def self.dump(db_opts, connect_opts = {})
+    uri = run_db_opt_for_new_uri(:dump, db_opts, connect_opts) do |database_opts|
+      PostgresAdmin.backup_pg_dump(database_opts)
+    end
+    _log.info("[#{db_opts[:dbname]}] database has been dumped up to file: [#{uri}]")
+    uri
+  end
 
+  # db_opts:
+  #   :dbname => 'vmdb_production',
+  #   :username => 'root',
+  #   :local_file => "/tmp/backup_1",      - Backup locally to the file specified
+
+  # connect_opts:
+  #   :uri => "smb://dev005.manageiq.com/share1",
+  #   :username => 'samba_one',
+  #   :password => 'Zug-drep5s',
+  #   :remote_file_name => "backup_1",     - Provide a base file name for the uploaded file
+  private_class_method def self.run_db_opt_for_new_uri(action, db_opts, connect_opts = {})
```

Comment:
This method name is not great. I would prefer we refactor this into more pieces to make the individual operations more obvious. For example, `with_mount_session` and `check_free_space`

Reply:
Cool, yeah, I can look into doing that.  I agree, the name sucks.

cc @jrafanie #namingIsHard #botherLJWheneverPossible

Reply:
I like naming things "run_things_and_stuff" to force myself to name it before merge.  Sometimes, I leave it like that so someone else can provide a better name.

Reply:
For me this is the only thing holding this PR up. @NickLaMuro are you going to try to refactor this or are you just looking for naming suggestions?

Reply:
@carbonin actively looking at this right now actually.

I think your naming suggestions are pretty solid, so probably just going to go with them, but I am noticing that half of what I would do for `with_mount_session` is also needed for the `EvmDatabaseOps.restore` method as well, so pondering how I can that in as well.

Since I am splitting things up a bit now (where before, it was just a copy-pasta for the most part), I am taking my time to do it right.  This is the disadvantage of going the full refactoring route, where I original just kinda did it half way.  Splitting this up via your suggestion is the overall better approach in my opinion, just has the cost of time to implement.

Reply:
Sounds good @NickLaMuro, just making sure :+1: